File: Resources/Scripter.AlternetUI/CallMethod.cs
using System;
using System.Diagnostics;
using Alternet.UI;
using Alternet.Drawing;

namespace ScriptSpace
{
    public class ScriptClass
    {
        static bool initialized;
        static SolidBrush arenaBackgroundBrush;
        static SolidBrush dotBrush;

        static double currentAngle;
        static double radius;
        static double dotRadius;
        static PointD center;

        static double DegreesToRadians(double degrees)
        {
            double radians = (Math.PI / 180) * degrees;
            return radians;
        }

        static void InitializeIfNeeded(RectD bounds)
        {
            if (initialized)
                return;

            arenaBackgroundBrush = new SolidBrush(Color.DarkBlue);
            dotBrush = new SolidBrush(Color.White);

            double maxSide = Math.Max(bounds.Width, bounds.Height);
            radius = (maxSide - (maxSide / 3)) / 2;
            dotRadius = maxSide / 20;

            center = new PointD(
                bounds.Left + bounds.Width / 2,
                bounds.Top + bounds.Height / 2);

            initialized = true;
        }

        public static void OnPaint(Graphics g, RectD bounds)
        {
            InitializeIfNeeded(bounds);

            var arenaBounds = bounds;
            arenaBounds.Inflate(-2, -2);
            g.FillEllipse(arenaBackgroundBrush, arenaBounds);

            var radians = DegreesToRadians(currentAngle);

            var dotCenter = new PointD(
                center.X + (int)(Math.Cos(radians) * radius),
                center.Y + (int)(Math.Sin(radians) * radius));

            g.FillEllipse(
                dotBrush,
                new RectD(
                    new PointD(dotCenter.X - dotRadius, dotCenter.Y - dotRadius), 
                    new SizeD(new PointD(dotRadius * 2, dotRadius * 2)))
                );
        }

        static double ConstrainAngle(double x)
        {
            x %= 360;
            if (x < 0)
                x += 360;

            return x;
        }

        public static void OnUpdate(int deltaTimeMs)
        {
            currentAngle += deltaTimeMs * 0.1;
            currentAngle = ConstrainAngle(currentAngle);
            Debug.WriteLine("Current Angle: " + currentAngle);
        }
        public static void Main()
        {
        }
    }
}

File: Resources/Scripter.AlternetUI/CallMethod.vb
Imports System
Imports System.Diagnostics
Imports Alternet.UI
Imports Alternet.Drawing

Namespace ScriptSpace
    Public Class ScriptClass
        Private Shared initialized As Boolean
        Private Shared arenaBackgroundBrush As SolidBrush
        Private Shared dotBrush As SolidBrush

        Private Shared currentAngle As Double
        Private Shared radius As Double
        Private Shared dotRadius As Double
        Private Shared center As PointD

        Private Shared Function DegreesToRadians(degrees As Double) As Double
            Dim radians As Double = (Math.PI / 180) * degrees
            Return radians
        End Function

        Private Shared Sub InitializeIfNeeded(bounds As RectD)
            If initialized Then
                Return
            End If

            arenaBackgroundBrush = New SolidBrush(Color.DarkBlue)
            dotBrush = New SolidBrush(Color.White)

            Dim maxSide As Double = Math.Max(bounds.Width, bounds.Height)
            radius = (maxSide - (maxSide / 3)) / 2
            dotRadius = maxSide / 20

            center = New PointD(
                bounds.Left + bounds.Width / 2,
                bounds.Top + bounds.Height / 2)

            initialized = True
        End Sub

        Public Shared Sub OnPaint(g As Graphics, bounds As RectD)
            InitializeIfNeeded(bounds)

            Dim arenaBounds As RectD = bounds
            arenaBounds.Inflate(-2, -2)
            g.FillEllipse(arenaBackgroundBrush, arenaBounds)

            Dim radians As Double = DegreesToRadians(currentAngle)

            Dim dotCenter As PointD = New PointD(
                center.X + CInt(Math.Cos(radians) * radius),
                center.Y + CInt(Math.Sin(radians) * radius))

            g.FillEllipse(
                dotBrush,
                New RectD(
                    New PointD(dotCenter.X - dotRadius, dotCenter.Y - dotRadius),
                    New SizeD(New PointD(dotRadius * 2, dotRadius * 2)))
                )
        End Sub

        Private Shared Function ConstrainAngle(x As Double) As Double
            x = x Mod 360
            If x < 0 Then
                x += 360
            End If

            Return x
        End Function

        Public Shared Sub OnUpdate(deltaTimeMs As Integer)
            currentAngle += deltaTimeMs * 0.1
            currentAngle = ConstrainAngle(currentAngle)
            Debug.WriteLine("Current Angle: " & currentAngle)
        End Sub

        Public Shared Sub Main()
        End Sub
    End Class
End Namespace